File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {ListNode}
     */
    reverseList(head: ListNode | null): ListNode {
        let current = head;
        let previous = null;
        while (head) {
            head = head.next;
            current.next = previous;
            previous = current;
            current = head;
        }
        return previous;
    }
}

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        Queue pq = new PriorityQueue<Number>();
        for (ListNode list: lists) {
            ListNode node = list;
            while (node != null) {
                pq.offer(node.val);
                node = node.next;
            }
        }
        if (pq.isEmpty()) {
            return null;
        }
        ListNode n = new ListNode();
        ListNode result = n;
        n.val = (int) pq.poll();

        while (pq.isEmpty() == false) {
            n.next = new ListNode();
            n = n.next;
            n.val = (int)pq.poll();
        }
        //n.next = null;
        return result;
    }
}
